match.py

```diff
@@ -26,92 +35,37 @@ def parse_args():
-                time.sleep(SLEEP_TIME)
-            result_json = household_project.get_results()
-            results = Results(c.systems, project_name, result_json)
-            results.insert_results(database.household_match_groups)
+def has_results_available(config):
+    available_results = set(map(lambda x: x.stem, Path(config.project_results_dir).glob('*.json')))
+    expected_results = set(config.projects)
+    if (available_results == expected_results):
+        return True
     else:
-        log.debug("Processing individuals")
-        if c.blocked:
-            log.debug("Blocked, extracting CLKs and blocks")
-            for system in c.systems:
-                c.extract_clks(system)
-                c.extract_blocks(system)
+        raise MissingResults(expected_results, available_results)
 
-        iter_num = 0
-        for project_name, schema in c.load_schema().items():
-            iter_num = iter_num + 1
-            project = Project(
-                project_name, schema, c.systems, c.entity_service_url, c.blocked
-            )
-            project.start_project()
-            for system in c.systems:
-                if c.blocked:
-                    project.upload_clks_blocked(
-                        system,
-                        c.get_clk(system, project_name),
-                        c.get_block(system, project_name),
-                    )
-                else:
-                    project.upload_clks(system, c.get_clks_raw(system, project_name))
-            project.start_run(c.matching_threshold)
-            running = True
-            print("\n--- RUNNING ---\n")
-            while running:
-                status = project.get_run_status()
-                print(status)
-                if status.get("state") == "completed":
-                    running = False
-                    break
-                time.sleep(SLEEP_TIME)
-            print("\n--- Getting results ---\n")
-            result_json = project.get_results()
-            results = Results(c.systems, project_name, result_json)
-            print(
-                "Matching groups for system "
-                + str(iter_num)
-                + " of "
-                + str(len(c.load_schema().items()))
-            )
-            results.insert_results(database.match_groups)
 
+def do_match(config):
+    has_results_available(config)
+    database = MongoClient(config.mongo_uri).linkage_agent
+    for file_name in Path(config.project_results_dir).glob('*.json'):
+        project_name = Path(file_name).stem
+        with open(file_name) as file:
+            result_json = json.load(file)
+            results = Results(config.systems, project_name, result_json)
+            print(f'Matching groups for project: {project_name}')
+            if config.household_match:
+                results.insert_results(database.match_groups)
+            else:
+                results.insert_results(database.house_hold_match_groups)
```

Comment:
Two quick notes here, 1 this appears to be reversed, and 2 for households the collection should be `household_match_groups`. 

More broadly though I think this check for household matching needs to happen either sooner, or also in `has_results_available`. For individual matching there are going to be multiple projects, for household matching there is only one. Trying household matching just now it's looking for the wrong thing, I get this error:

```
__main__.MissingResults: Missing results for projects: name-sex-dob-zip, name-sex-dob-phone, name-sex-dob-addr
```

Reply:
Good catch. Should be resolved by https://github.com/mitre/linkage-agent-tools/pull/21/commits/5c7d0f71a0d1a833900c1ded39bba22ebe02aa3a